src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -302,4 +311,20 @@ public String findStatus(UUID resultUuid) {
             return null;
         }
     }
+
+    private Pageable addDefaultSort(Pageable pageable, String defaultSortColumn) {
+        if (pageable.isPaged()) {
+            if (pageable.getSort().equals(Sort.unsorted())) {
+                //if there is no sort on the original request we just add the default one
+                return PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.by(DEFAULT_SORT_DIRECTION, defaultSortColumn));
+            } else {
+                //else we restore the original sort, and then we add the default sort on top of it to ensure there is a deterministic result
+                Sort additionalSort = Sort.by(new Sort.Order(DEFAULT_SORT_DIRECTION, defaultSortColumn));
+                Sort finalSort = pageable.getSort().and(additionalSort);
```

Comment:
You add it here even if a sort already exists in `pageable` on `defaultSortColumn`, it will be there 2 times, maybe with both directions... not a good idea

Reply:
fixed